core/iwasm/fast-jit/cg/x86-64/jit_codegen_x86_64.cpp

```diff
@@ -1174,8 +1210,7 @@ mov_imm_to_r_i64(x86::Assembler &a, int32 reg_no, int64 data)
 static bool
 mov_r_to_r_i64(x86::Assembler &a, int32 reg_no_dst, int32 reg_no_src)
 {
-    if (reg_no_dst != reg_no_src)
-        a.mov(regs_i64[reg_no_dst], regs_i64[reg_no_src]);
+    a.mov(regs_i64[reg_no_dst], regs_i64[reg_no_src]);
```

Comment:
Should not remove `if (reg_no_dst != reg_no_src)`, or it might impact the performance